### ISSUE TITLE
[7.1.0] long waiting times when yield gets detected.

### ISSUE THREAD

**Issue by hilfazer** [bitwes/Gut#288]
This test finishes instantly in GUT 7.0.0 (0.0s) but takes 2.0s in GUT 7.1.0.

```
extends "res://addons/gut/test.gd"

func test_yield1():
	yield( get_tree(), "idle_frame" )
	assert_true(true)

func test_yield2():
	yield( get_tree(), "idle_frame" )
	assert_true(true)

func test_yield3():
	yield( get_tree(), "idle_frame" )
	assert_true(true)

func test_yield4():
	yield( get_tree(), "idle_frame" )
	assert_true(true)

func test_yield5():
	yield( get_tree(), "idle_frame" )
	assert_true(true)

```

```

Godot version:  3.3.0
GUT version:  7.1.0


res://tests/Test_many_yields.gd
  Only running tests like: "test_"
* test_yield3
    -- Yield detected, waiting --
* test_yield1
    -- Yield detected, waiting --
* test_yield4
    -- Yield detected, waiting --
* test_yield2
    -- Yield detected, waiting --
* test_yield5
    -- Yield detected, waiting --
5/5 passed.


*** Run Summary ***
All tests passed

Totals
Scripts:          1
Tests:            5
Passing asserts:  5
Failing asserts:  0
Pending:          0

Warnings/Errors:
* 2 Warnings.


5 passed 0 failed.  Tests finished in 2.0s
```

And this one takes 0.1s for 7.0.0 and 0.4s for 7.1.0
```
extends "res://addons/gut/test.gd"


func test_yield():
	yield(yield_for(.1), YIELD)
```

```
Godot version:  3.3.0
GUT version:  7.1.0


res://tests/Test_yield.gd
  Only running tests like: "test_"
* test_yield
    -- Yielding (0.1s) --
    [WARNING]:  Test did not assert


*** Run Summary ***
All tests passed

Totals
Scripts:          1
Tests:            1
Passing asserts:  0
Failing asserts:  0
Pending:          0

Warnings/Errors:
* 3 Warnings.


0 passed 0 failed.  Tests finished in 0.4s
```

**Comment by afonsolage** [comment]
Same issue here. There is any workaround?

**Comment by bitwes** [comment]
I'm not sure the exact cause yet, but if you decrease the time GUT waits when it detects a yield you can increase the speed.  I tested `0.1` which dropped the time to execute the example tests from `2.5` to `1.0`.

I don't think this is the cause since it greatly speeds up the "waiting" animation, but it will get you by in the short term.

Here's the line:  https://github.com/bitwes/Gut/blob/3e360286518e1c797fb7c79999433ed5915328f9/addons/gut/gut.gd#L586

**Comment by afonsolage** [comment]
Indeed, after understand this part of the code, I set it to 50ms, since it only to give time (yield) of other routines to run.

```
_wait_timer.set_wait_time(0.05)
```

it reduced the time from 4s (I have many `yield(yield_to(get_tree(), "idle_frame", 1), YIELD)`) to 2s, which is acceptable. But it would be nice if it was 0.00, as mentioned by @hilfazer

**Comment by hilfazer** [comment]
I tried GUT's `yield_for()` and `yield_to()`  and they're as slow.

**Comment by bitwes** [comment]
I think this is fixed in 7.2.0.  It now yields 2 frames between tests which should speed things up.   Can anyone verify?

**Comment by hilfazer** [comment]
I'm getting same time with 7.2.0. Tried with Godot 3.3.2 and Godot 3.4.

BTW the new GUT icon looks absolutely scary.
